### PR TITLE
Fixes #33823 - job wizard submit data and create a job

### PR DESCRIPTION
**Pull request by MariaAga**
- [x] need to add redirect and toast after submit as user gets no indication currently

### CODE REVIEW

**Comment by MariaAga** [comment]
@laviro can you take a look?
It's missing tests but I'm not sure what to test in this pr

File: webpack/JobWizard/submit.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      password,
+      key_passphrase: keyPassphrase,
+      recurrence:
+        repeatType === repeatTypes.noRepeat
+          ? null
+          : {
+              cron_line: getCronLine(),
+              max_iteration: repeatAmount,
+              end_time: ends.length ? ends.toISOString() : null,
+              purpose: '', // TODO add value after https://projects.theforeman.org/issues/33814
```
Comment (MariaAga) [inline webpack/JobWizard/submit.js:72]: https://github.com/theforeman/foreman_remote_execution/pull/666

File: webpack/JobWizard/submit.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            },
+      scheduling: {
+        start_at: starts.length ? starts.toISOString() : null,
```
Comment (adamruzicka) [inline webpack/JobWizard/submit.js:75]: Would it be possible to not include the `scheduling` and `recurrence` keys in the request if the job should execute immediately? The api is confused by this and sets the job to execute in the future at time of the request, which kinda does the same thing but is less efficient than just starting it right away. An alternative would be addressing this in the api

File: webpack/JobWizard/steps/HostsAndInputs/index.js [hunk 3]
```diff
@@ -74,12 +56,13 @@ const HostsAndInputs = ({
   const [hostPreviewOpen, setHostPreviewOpen] = useState(false);
   useEffect(() => {
     debounce(() => {
+      console.log(buildHostQuery(selected, hostsSearchQuery));
```
Comment (MariaAga) [inline webpack/JobWizard/steps/HostsAndInputs/index.js:59]: leftover to delete

**Review by adamruzicka — COMMENTED** [review]
If I get to the last page, kill the server and click submit, nothing happens. There are some `Uncaught (in promise) TypeError: _ref5.response is undefined` errors in the browser console, but nothign immediately visible to the user.

**Comment by MariaAga** [comment]
I think this kind of error will usually not be visible to the user, but I added a few fallbacks for the toast message

**Review by laviro — COMMENTED** [review]
Code wise looks good,
when clicking on the last stage "Run" I get the following error:
```
Uncaught TypeError: starts.toISOString is not a function
    at submit (submit.js:76)
    at onSave (JobWizard.js:194)
    at Wizard.onNext (Wizard.js:34)
    at HTMLUnknownElement.callCallback (react-dom.development.js:188)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:237)
    at invokeGuardedCallback (react-dom.development.js:292)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:306)
    at executeDispatch (react-dom.development.js:389)
    at executeDispatchesInOrder (react-dom.development.js:414)
    at executeDispatchesAndRelease (react-dom.development.js:3278)
```

**Comment by MariaAga** [comment]
Thanks @laviro!
Added a wrap of `new Date()` around the end and start time to fix that

**Comment by laviro** [comment]
should we do the same in the `starts before field`: https://github.com/theforeman/foreman_remote_execution/pull/660

**Review by laviro — APPROVED** [review]

**Comment by MariaAga** [comment]
I added the use of `submitStatus` so if a req has started the user won't be able to click submit again. should a user be able to submit the same job wizard page again if an error accrued? @adamruzicka

**Comment by adamruzicka** [comment]
> should a user be able to submit the same job wizard page again if an error accrued?

In general yes, unless the server indicated there is something wrong with the user's input, which hopefully should not happen.

**Comment by MariaAga** [comment]
I see, so changed it to `isSubmitting` so if the request is running or finished successfully the submit is disabled

**Comment by adamruzicka** [comment]
I'm getting this if I only select a host, fill in the command to run and leave everything else untouched.

image

**Comment by MariaAga** [comment]
Added `start_at` `start_before` and `purpose`

**Comment by adamruzicka** [comment]
This is still valid https://github.com/theforeman/foreman_remote_execution/pull/662#pullrequestreview-797429907 . We can now handle when the server responds nicely (even if with error), but if the server is down completely, it still fails silently

image

**Comment by adamruzicka** [comment]
In chrome 95 I get this 
image

In firefox 92.0 I get:
image

**Comment by MariaAga** [comment]
I couldn't reproduce it, but added a conditional chain `response?.__` so it hopefully won't fail

**Comment by adamruzicka** [comment]
Looks better now, lets' get it in.

**Review by adamruzicka — APPROVED** [review]
Works nicely

**Comment by adamruzicka** [comment]
Thank you @MariaAga & @Ron-Lavi !

**Pull request merged**